packages/components/src/components/form/Switch/Switch.tsx

```diff
@@ -71,52 +80,79 @@ const CheckboxInput = styled.input`
     width: 1px;
 `;
 
+const Label = styled.div<Pick<SwitchProps, 'isDisabled' | 'isAlert' | 'isSmall'>>`
+    color: ${({ isAlert, isDisabled, theme }) =>
+        getLabelColor(theme, { alert: isAlert, disabled: isDisabled })};
+    text-align: left;
+    ${({ isSmall }) => (isSmall ? typography.label : typography.body)}
+`;
+
 export interface SwitchProps {
     id?: string;
     isChecked: boolean;
+    label?: string;
     onChange: (isChecked?: boolean) => void;
     isDisabled?: boolean;
-    isSmall?: boolean;
+    isAlert?: boolean;
+    isSmall?: boolean; // TODO: legacy prop
     className?: string;
     dataTest?: string;
+    labelPosition?: 'left' | 'right';
 }
 
 export const Switch = ({
     onChange,
     id,
```

Comment:
i'd remove this prop and use `getWeakRandomId()` locally to get an ID and pass it to the label, what do you think?

Reply:
lmk if we understand right each other 9ada0fde38e5e788ad4b7eba036e50f07baef91f